src/evm/protocol/vm/state.rs

```diff
@@ -126,6 +135,34 @@ where
         Ok(())
     }
 
+    /// Sets the spot prices for a pool for all possible pairs of the given tokens.
```

Comment:
maybe here we can also add some advice on usage?
like that if the use is interested in fetching really fast prices it would be nice to set the spot prices on the pool every time the pool actually changes and not only when the user needs the prices